esp/esdllib/esdl_script.cpp

```diff
@@ -130,51 +202,312 @@ class CEsdlTransformOperationVariable : public CEsdlTransformOperationBase
+                    if (streq(op, "http-header"))
+                        m_headers.append(*new CEsdlTransformOperationHttpHeader(xpp, stag, prefix));
+                    else if (streq(op, "content"))
+                        m_content.setown(new CEsdlTransformOperationHttpContentXml(xpp, stag, prefix));
+                    else
+                        xpp.skipSubTreeEx();
+                    break;
+                }
+                case XmlPullParser::END_TAG:
+                case XmlPullParser::END_DOCUMENT:
+                    return;
+            }
+        }
+    }
+
+    virtual ~CEsdlTransformOperationHttpPostXml()
+    {
+    }
+
+    void buildHeaders(IEsdlScriptContext * scriptContext, IXpathContext * targetContext, IXpathContext * sourceContext, IProperties *headers)
+    {
+        if (!m_headers.length())
+            return;
+        CXpathContextLocation location(targetContext);
+        targetContext->addElementToLocation("headers");
+        ForEachItemIn(i, m_headers)
+            m_headers.item(i).processHeader(scriptContext, targetContext, sourceContext, headers);
+    }
+
+    void buildRequest(IEsdlScriptContext * scriptContext, IXpathContext * targetContext, IXpathContext * sourceContext, const char *url, IProperties *headers)
+    {
+        CXpathContextLocation location(targetContext);
+        targetContext->addElementToLocation("request");
+        targetContext->ensureSetValue("@url", url, true);
+        buildHeaders(scriptContext, targetContext, sourceContext, headers);
+        if (m_content)
+            m_content->process(scriptContext, targetContext, sourceContext);
+    }
+
+    void postRequest(IEsdlScriptContext * scriptContext, IXpathContext * targetContext, IXpathContext * sourceContext, const char *url, IProperties *headers)
+    {
+        VStringBuffer xpath("/esdl_script_context/%s/%s/request/content/*[1]", m_section.str(), m_name.str());
+
+        StringBuffer content;
+        sourceContext->toXml(xpath, content);
+        if (!content)
+            return;
+        CXpathContextLocation location(targetContext);
+        targetContext->addElementToLocation("response");
+
+        Owned<IHttpClientContext> httpCtx = getHttpClientContext();
+        Owned<IHttpClient> httpclient = httpCtx->createHttpClient(NULL, url);
+        if (!httpclient)
+            return;
+
+        try
+        {
+            StringBuffer status;
+            StringBuffer response;
+            httpclient->sendRequest(headers, "POST", "text/xml", content, response, status);
+            targetContext->ensureSetValue("@status", status.str(), true);
+            size32_t len = response.trim().length();
+            if (!len)
+            {
+                throw MakeStringException(ESDL_SCRIPT_Error, "ESDL Script empty result calling http-post-xml %s url(%s)", m_traceName.str(), url);
+            }
+            else if (*response.str()=='<')
```

Comment:
I don't know if we really can or need to do anything differently here, but does the HTTP spec say anything about guarantees of response type given a certain Content Type? I'm guessing not really, but we could have a situation here where some text is returned and just happens to start with '<'. Then we'd just return whatever XML parse error resulting from addXmlContent().